searx/search/__init__.py

```diff
@@ -97,6 +97,10 @@ def _get_requests(self):
             if request_params is None:
                 continue
 
+            # stop request if it exceeds engine's rate limit
+            if processor.exceeds_rate_limit():
```

Comment:
I suggest to move this call to [extend_container_if_suspended](https://github.com/searxng/searxng/blob/c17c7dc0504daa6be675de6ac08fed03ef3246f7/searx/search/processors/abstract.py#L132-L138), so the user can get a message.

Something like this:
```python
    def extend_container_if_suspended(self, result_container):
        if self.suspended_status.is_suspended:
            result_container.add_unresponsive_engine(
                self.engine_name, self.suspended_status.suspend_reason, suspended=True
            )
            return True
        if self.exceeds_rate_limit():
            result_container.add_unresponsive_engine(
                self.engine_name, "Exceed rate limit", suspended=True
            )
            return True
        return False
```